### DISCUSSION TITLE
Load gradient for non-Cartesian trajectory from text file

### DISCUSSION THREAD

**Discussion by Michaelsqj** [pulseq/pulseq#45] [General]
Hi there,

Is there any way that I could load arrays of gradient from text file to the sequence on the scanner?

Yours,
Qijia

**Comment by QingpingChen1201** [comment]
Hi Qijia,
Thanks for your interest in Pulseq! For the implementation of arbitrary gradients using Pulseq, I would recommend you have a look at the "12_Radial_and_nonCartesian" topic of our Pulseq tutorial (https://github.com/pulseq/tutorials/tree/main/12_Radial_and_nonCartesian). Please do not hesitate to contact us if you have any problems or need any further information.
Best regards,
Qingping

**Comment by m-a-x-i-m-z** [comment]
Hi Qijia, 

you can load the desired shape from a text file in Matlab or Python and generate a Pulseq sequence based on that. Qinping's response hints towards that. However, there is no way that you sit at the scanner and have Pulseq sequence loading your gradient file directly. Hope this together with the above reply answers your question. 

Best regards, 
Maxim

**Comment by Michaelsqj** [comment]
Hi Maxim,Qinqing,

Thank you very much for your reply. May I ask an additional question about how many different trajectories could be loaded to matlab and compiled in one single sequence? In our case, we might have thousands of trajectories in total.

Yours,
Qijia

**Comment by m-a-x-i-m-z** [comment]
Hi Qijia, 

the answer to your question is scanner dependent and is limited by the available shape memory, which we don't control at this point. So sequences either run or they don't on the scanner and you can experiment with the number of shapes that your are able to load while at the scanner. From experience our older Trio Tim is able to only load dozens to hundreds of shapes, whereas Prisma Fit can load hundreds to thousands, but of course both the shape durations and the number of shapes are important. 

One thing: per default the Pulseq interpreter enables libBalance for all sequences, but it is known to crash for longer sequences, so you may have to disable it to be able to run longer sequences. 

One more thing: if your sequence gets bigger than 50-60 Mb, the user interface may become very sluggish (and libBalance will certainly crash) but I've also succeeded in running sequences as large as 180 Mb (which did include  waiting for ~5 minutes for the sequences to load :) 

Best regards, 
Maxim
